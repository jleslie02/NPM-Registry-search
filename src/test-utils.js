import React from "react";
import { render } from "react-testing-library";
import makeAtomicTheme from "./theme";

export const themedRender = (MyComponent, props) => {
  return render(<MyComponent theme={makeAtomicTheme()} {...props} />);
};

export const themedComponent = (MyComponent, props) => {
  return <MyComponent theme={makeAtomicTheme()} {...props} />;
};
