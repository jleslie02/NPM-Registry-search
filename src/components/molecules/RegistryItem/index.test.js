// Static component with conditional rendering so snapshot test should suffice
import renderer from "react-test-renderer";
import RegistryIem from "./index";
import { themedComponent } from "../../../test-utils";

it("renders correctly with default props", () => {
  const props = {
    score: {},
    flags: null,
    data: {}
  };
  const tree = renderer.create(themedComponent(RegistryIem, props)).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with partial props", () => {
  const props = {
    score: {
      detail: {
        maintenance: 0.9,
        quality: 0.9,
        popularity: 0.9
      }
    },
    data: {
      name: "a package"
    }
  };
  const tree = renderer.create(themedComponent(RegistryIem, props)).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with full props", () => {
  const props = {
    score: {
      detail: {
        maintenance: 0.9,
        quality: 0.9,
        popularity: 0.9
      }
    },
    flags: {
      unstable: true
    },
    data: {
      name: "a package"
    }
  };
  const tree = renderer.create(themedComponent(RegistryIem, props)).toJSON();
  expect(tree).toMatchSnapshot();
});
