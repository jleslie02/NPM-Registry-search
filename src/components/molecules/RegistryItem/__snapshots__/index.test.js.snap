// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default props 1`] = `
<div
  className="registryItem css-e1h06c-RegistryItem"
  data-testid="registry-item"
>
  <div
    className="css-1wumyry-RegistryItem"
  >
    u
  </div>
  <div
    className="css-chobbe-RegistryItem"
  >
    <div>
      <div
        className="css-kk417u-RegistryItem"
        onClick={[Function]}
      >
        <span />
        <div />
      </div>
      <div
        className="css-o0dv41-RegistryItem"
      >
        <div
          className="user"
        >
          <span
            className="fa fa-user"
          />
          unknown
        </div>
      </div>
      <div
        className="css-czi24t-RegistryItem"
      />
    </div>
  </div>
  <div
    className="css-1966wl8-RegistryItem"
  />
  <div
    className="css-fuf05g-RegistryItem"
  />
</div>
`;

exports[`renders correctly with full props 1`] = `
<div
  className="registryItem css-e1h06c-RegistryItem"
  data-testid="registry-item"
>
  <div
    className="css-1wumyry-RegistryItem"
  >
    u
  </div>
  <div
    className="css-chobbe-RegistryItem"
  >
    <div>
      <div
        className="css-kk417u-RegistryItem"
        onClick={[Function]}
      >
        <span>
          a package
        </span>
        <div />
      </div>
      <div
        className="css-o0dv41-RegistryItem"
      >
        <div
          className="user"
        >
          <span
            className="fa fa-user"
          />
          unknown
        </div>
      </div>
      <div
        className="css-czi24t-RegistryItem"
      >
        <div
          className="statWrapper"
        >
          <span
            className="maintenance statName"
          >
            maintenance
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
          <div
            className="css-cmd4p3"
          />
        </div>
        <div
          className="statWrapper"
        >
          <span
            className="quality statName"
          >
            quality
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
          <div
            className="css-cmd4p3"
          />
        </div>
        <div
          className="statWrapper"
        >
          <span
            className="popularity statName"
          >
            popularity
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1966wl8-RegistryItem"
  >
    <span
      className="css-o20i0f"
    >
      unstable
    </span>
  </div>
  <div
    className="css-fuf05g-RegistryItem"
  />
</div>
`;

exports[`renders correctly with partial props 1`] = `
<div
  className="registryItem css-e1h06c-RegistryItem"
  data-testid="registry-item"
>
  <div
    className="css-1wumyry-RegistryItem"
  >
    u
  </div>
  <div
    className="css-chobbe-RegistryItem"
  >
    <div>
      <div
        className="css-kk417u-RegistryItem"
        onClick={[Function]}
      >
        <span>
          a package
        </span>
        <div />
      </div>
      <div
        className="css-o0dv41-RegistryItem"
      >
        <div
          className="user"
        >
          <span
            className="fa fa-user"
          />
          unknown
        </div>
      </div>
      <div
        className="css-czi24t-RegistryItem"
      >
        <div
          className="statWrapper"
        >
          <span
            className="maintenance statName"
          >
            maintenance
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
          <div
            className="css-cmd4p3"
          />
        </div>
        <div
          className="statWrapper"
        >
          <span
            className="quality statName"
          >
            quality
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
          <div
            className="css-cmd4p3"
          />
        </div>
        <div
          className="statWrapper"
        >
          <span
            className="popularity statName"
          >
            popularity
          </span>
          <span
            className="statNumber"
          >
            90
            %
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1966wl8-RegistryItem"
  />
  <div
    className="css-fuf05g-RegistryItem"
  />
</div>
`;
